hudi-spark-datasource/hudi-spark/src/test/java/org/apache/hudi/functional/TestMORColstats.java

```diff
@@ -237,24 +232,27 @@ private void testBaseFileAndLogFileUpdateMatchesHelper(Boolean shouldAsyncCompac
     doWrite(updatedRecord);
     if (shouldRollback) {
       deleteLatestDeltacommit();
+      enableInlineCompaction(shouldInlineCompact);
```

Comment:
That's how I was originally tying to do it, but the behavior with the rollback was strange. If I set `hoodie.compact.inline.max.delta.commits` to 4 the compaction was not happening, but if I set it to 3 it would happen too early. So that's why I set the value to 1 and do it right before the last write

Reply:
It's because we rollback the 3rd commit, so we want it to compact on the second 3rd commit so that's why we enable it just before